.github/workflows/code-quality.yml

```
@@ -0,0 +1,196 @@
+        with:
+          php-version: ${{ matrix.php-version }}
+          extensions: mbstring, xml, json, tokenizer, bcmath, sqlite3
+          coverage: xdebug
+          tools: composer:v2, phpcs, phpstan, psalm
+      
+      - name: Get Composer cache directory
+        id: composer-cache
+        run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
+      
+      - name: Cache Composer dependencies
+        uses: actions/cache@v3
```

actionlint (1.7.7)

the runner of "actions/cache@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)